src/grpcbox_subchannel.erl

```diff
@@ -14,7 +16,8 @@
          ready/3,
          disconnected/3]).
 
--record(data, {endpoint :: grpcbox_channel:endpoint(),
+-record(data, {name :: any(),
```

Comment:
The `name` field does not need to be introduced as it is already set with the endpoint in the upstream implementation of grpcbox. The code here is confusing.

Reply:
grpcbox_subchannel contains name, endpoint, and channel properties, The gproc_pool:connect_worker function is called during grpcbox_subchannel initialization with the channnel name parameter. The grpcbox_subchannel terminal calls gproc_pool:disconnect_worker and gproc_pool:remove_worker with channel and endpoint arguments, which are errors, Use the channel and endpoint parameters correctly.